cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -324,15 +334,31 @@ private List<LocalFile> getRuntimeLocalFiles(Collection<? extends LocalFile> run
                                                File tempDir) throws Exception {
     LocationFactory locationFactory = new LocalLocationFactory(tempDir);
     List<LocalFile> localFiles = new ArrayList<>(runtimeLocalFiles);
-    localFiles.add(DataprocJarUtil.getTwillJar(locationFactory));
-    localFiles.add(DataprocJarUtil.getLauncherJar(locationFactory));
+    localFiles.add(getTwillJar(locationFactory));
+    localFiles.add(getLauncherJar(locationFactory));
 
     // Sort files in descending order by size so that we upload concurrently large files first.
     localFiles.sort(Comparator.comparingLong(LocalFile::getSize).reversed());
 
     return localFiles;
   }
 
+  private LocalFile getTwillJar(LocationFactory locationFactory) throws IOException {
```

Comment:
It looks like DataprocJarUtil.getTwillJar() already checks the location, are these methods needed?